websockets/websockets.go

```diff
@@ -586,6 +613,55 @@ func (w *webSocket) send(msg goja.Value) {
 	}
 }
 
+// Ping sends a ping message over the websocket.
+func (w *webSocket) ping() {
+	deadline := time.Now().Add(writeWait)
+	pingID := strconv.Itoa(w.sendPings.counter)
+
+	data := []byte(pingID)
+
+	err := w.conn.WriteControl(websocket.PingMessage, data, deadline)
+	if err != nil {
+		_ = w.callErrorListeners(err)
+		return
+	}
+
+	w.sendPings.timestamps[pingID] = time.Now()
+	w.sendPings.counter++
+}
+
+func (w *webSocket) trackPong(pingID string) {
+	pongTimestamp := time.Now()
+
+	pingTimestamp, ok := w.sendPings.timestamps[pingID]
+	if !ok {
+		// We received a pong for a ping we didn't send; ignore
+		// (this shouldn't happen with a compliant server)
+		return
+	}
+
+	metrics.PushIfNotDone(w.vu.Context(), w.vu.State().Samples, metrics.Sample{
+		TimeSeries: metrics.TimeSeries{
+			Metric: w.builtinMetrics.WSPing,
+			Tags:   w.tagsAndMeta.Tags,
+		},
+		Time:     pongTimestamp,
+		Metadata: w.tagsAndMeta.Metadata,
+		Value:    metrics.D(pongTimestamp.Sub(pingTimestamp)),
+	})
+}
+
+// isStateOpen checks if the websocket is in the OPEN state
+// otherwise it throws an error (panic)
+func (w *webSocket) isStateOpen() {
```

Comment:
maybe 'assertStateOpen' 


with a name like `isStateOpen` I expect it will give me back a boolean